Source/Global/global.h

```diff
@@ -55,6 +55,9 @@ typedef struct http_singleton
     uint32_t m_timeoutWindowInSeconds;
     uint32_t m_retryDelayInSeconds;
 
+    // Platform-specific context for calls
+    std::shared_ptr<HCPlatformContext> m_platformContext;
```

Comment:
You can't own the context the clients will pass in.
(We may need to track both the current perform context and the "default" context separately for cleanup.

I'm not sure why you are using a shared pointer either since you never have more than one copy

Reply:
Exposing the context to clients feels really wrong here. I think the http call has access to the singleton struct and it probably makes more sense to reference it directly.

---
In reply to: [190330199](https://github.com/Microsoft/libHttpClient/pull/165#discussion_r190330199) [](ancestors = 190330199)